File: web/src/main/java/cz/inovatika/arup/digiarchiv/web/index/models/Komponenta.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package cz.inovatika.arup.digiarchiv.web.index.models;

import com.alibaba.fastjson.JSON;
import java.lang.annotation.Annotation;
import java.util.Map;
import java.util.logging.Level;
import java.util.logging.Logger;
import org.apache.commons.csv.CSVRecord;
import org.apache.solr.client.solrj.beans.Field;

/**
 *
 * @author alberto
 */
public class Komponenta {

  static final Logger LOGGER = Logger.getLogger(Komponenta.class.getName());

  @Field
  public String ident_cely;

  @Field
  public String parent;

  @Field
  public String obdobi;

  @Field
  public String jistota;

  @Field
  public String presna_datace;

  @Field
  public String areal;

  @Field
  public int aktivita_sidlistni;

  @Field
  public int aktivita_pohrebni;

  @Field
  public int aktivita_vyrobni;

  @Field
  public int aktivita_tezebni;

  @Field
  public int aktivita_kultovni;

  @Field
  public int aktivita_komunikace;

  @Field
  public int aktivita_deponovani;

  @Field
  public int aktivita_intruze;

  @Field
  public int aktivita_boj;

  @Field
  public int aktivita_jina;

  @Field
  public String poznamka;

  @Field
  public String nalez;

  public static Komponenta fromCSV(Map<String, Integer> header, CSVRecord record) {

    Komponenta ret = new Komponenta();
    Class aClass = Komponenta.class;
    java.lang.reflect.Field[] fields = aClass.getDeclaredFields();

    for (java.lang.reflect.Field field : fields) {
      if (header.containsKey(field.getName())) {
        try {
          field.set(ret, record.get(header.get(field.getName())));
        } catch (IllegalArgumentException | IllegalAccessException ex) {
          LOGGER.log(Level.SEVERE, null, ex);
        }
      }
    }
    return ret;
  }
  
  @Override
  public String toString() {
    return JSON.toJSONString(this);
  }

}

File: web/src/main/ng/src/app/shared/komponenta.ts
export class Komponenta {
  idx: number;
  ident_cely: string;
  parent: string;
  poradi:string;
  
  poznamka:string;

  obdobi:string;
  obdobi_poradi:string;
  jistota:string;
  presna_datace:string;
  areal:string;
  aktivita_sidlistni:boolean;
  aktivita_tezebni:boolean;
  aktivita_vyrobni:boolean;
  aktivita_komunikace:boolean;
  aktivita_boj:boolean;
  aktivita_kultovni:boolean;
  aktivita_pohrebni:boolean;
  aktivita_deponovani:boolean;
  aktivita_intruze:boolean;
  aktivita_jina:boolean;
  
  setFieldFromDokument(dk: any, idx: string){
    
    this.ident_cely = dk['komponenta_dokumentu_ident_cely'][idx];
    this.parent = dk['komponenta_dokumentu_parent'][idx];
    this.poradi = dk['komponenta_dokumentu_poradi'][idx];
    this.poznamka = dk['komponenta_dokumentu_poznamka'][idx];
    this.obdobi = dk['komponenta_dokumentu_obdobi'][idx];
    this.obdobi_poradi = dk['komponenta_dokumentu_obdobi_poradi'][idx];
    this.jistota = dk['komponenta_dokumentu_jistota'][idx];
    this.presna_datace = dk['komponenta_dokumentu_presna_datace'][idx];
    this.areal = dk['komponenta_dokumentu_areal'][idx];
    this.aktivita_sidlistni = dk['komponenta_dokumentu_aktivita_sidlistni'][idx] === "1";
    this.aktivita_tezebni = dk['komponenta_dokumentu_aktivita_tezebni'][idx] === "1";
    this.aktivita_vyrobni = dk['komponenta_dokumentu_aktivita_vyrobni'][idx] === "1";
    this.aktivita_komunikace = dk['komponenta_dokumentu_aktivita_komunikace'][idx] === "1";
    this.aktivita_boj = dk['komponenta_dokumentu_aktivita_boj'][idx] === "1";
    this.aktivita_kultovni = dk['komponenta_dokumentu_aktivita_kultovni'][idx] === "1";
    this.aktivita_pohrebni = dk['komponenta_dokumentu_aktivita_pohrebni'][idx] === "1";
    this.aktivita_deponovani = dk['komponenta_dokumentu_aktivita_deponovani'][idx] === "1";
    this.aktivita_intruze = dk['komponenta_dokumentu_aktivita_intruze'][idx] === "1";
    this.aktivita_jina = dk['komponenta_dokumentu_aktivita_jina'][idx] === "1";
  
  }
}
